### PR TITLE
Update `time` to use SysTick

### PR DESCRIPTION
**Pull request by tannewt**
This doesn't actually use ASF4 at all but enables it again. This depends on #258

### CODE REVIEW

File: atmel-samd/tick.c [hunk 1]
```diff
@@ -29,18 +44,21 @@ void timer_tick(const struct timer_task *const timer_task) {
 }
 
 void tick_init() {
-    #ifdef SAMD21
-    _pm_enable_bus_clock(PM_BUS_APBC, TC5);
-    #endif
-    // _gclk_enable_channel(TC5_GCLK_ID, GCLK_SOURCE_DFLL48M);
-
-    // timer_init(&ms_timer, TC5, _tc_get_timer());
+    SysTick->LOAD = ((CONF_CPU_FREQUENCY / 1000 - 1) << SysTick_LOAD_RELOAD_Pos);
```
Comment (dhalbert) [inline atmel-samd/tick.c]: Could you add a TODO to replace this eventually, or have this call a HAL routine that returns the frequency?  Could call `common_hal_mcu_processor_get_frequency()`.

File: atmel-samd/tick.c [hunk 2]
```diff
@@ -29,18 +44,21 @@ void timer_tick(const struct timer_task *const timer_task) {
...
-    // task.mode     = TIMER_TASK_REPEAT;
-    // timer_add_task(&ms_timer, &task);
-    //
-    // timer_start(&ms_timer);
+void tick_delay(uint32_t us) {
+    uint32_t fractional = SysTick->VAL;
+    uint64_t start_ms = ticks_ms;
+    while (us > 1000) {
+        while (ticks_ms == start_ms) {}
+        us -= fractional;
```
Comment (dhalbert) [inline atmel-samd/tick.c]: Based on my Googling, `SysTick` units are dependent on clock frequency, right? The SysTick timer is decremented on each clock cycle. So for instance at 48MHz, one tick is 1sec/48e6. So are you mixing microsoeconds (`us`) with SysTicks units?

Comment (tannewt) [reply atmel-samd/tick.c]: Yes! You are right.

**Comment by tannewt** [comment]
Thanks for the review! I'll get to this in a few days after Maker Faire craziness.

**Comment by tannewt** [comment]
@dhalbert could you take another look at this today? Thanks!

File: atmel-samd/tick.c [hunk 3]
```diff
@@ -29,18 +45,23 @@ void timer_tick(const struct timer_task *const timer_task) {
...
 
 void tick_init() {
-    #ifdef SAMD21
-    _pm_enable_bus_clock(PM_BUS_APBC, TC5);
-    #endif
-    // _gclk_enable_channel(TC5_GCLK_ID, GCLK_SOURCE_DFLL48M);
-
-    // timer_init(&ms_timer, TC5, _tc_get_timer());
+    uint32_t ticks_per_ms = common_hal_mcu_processor_get_frequency() / 1000;
+    SysTick->LOAD = ((ticks_per_ms - 1) << SysTick_LOAD_RELOAD_Pos);
```
Comment (dhalbert) [inline atmel-samd/tick.c]: You could do `SysTick_Config(ticks_per_ms)` to replace lines 49-53.

I remembered what Dean told me about CMSIS and thought there might be a function to do this.

**Comment by tannewt** [comment]
Ok, this is ready for review. I confirmed `time.sleep` works. I don't actually know what code calls delay.

**Pull request merged**